lib/EpdFont/scripts/build-sd-fonts.py

```
@@ -0,0 +1,331 @@
+DOWNLOAD_DIR = SCRIPT_DIR / "downloaded_fonts"
+INSTANCE_DIR = SCRIPT_DIR / "instanced_fonts"
+
+
+def download_font(url: str, dest: Path) -> Path:
+    """Download a font file if not already cached. Returns the local path."""
+    if dest.exists():
+        return dest
+    dest.parent.mkdir(parents=True, exist_ok=True)
+    print(f"  Downloading {dest.name}...")
+    try:
+        urllib.request.urlretrieve(url, dest)
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)